### PR TITLE
[no-release-notes] add immutable chunker

### PR DESCRIPTION
**Pull request by max-hoffman**

### CODE REVIEW

File: go/store/prolly/tree/immutable_tree.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	"io"
+
+	"github.com/dolthub/dolt/go/store/prolly/message"
+)
+
+const defaultFixedChunkLength = 4000
+
+var ErrInvalidChunkSize = errors.New("invalid chunkSize; value must be > 1")
+
+// buildImmutableTree writes the contexts of |reader| as an append-only
```
Comment (andrew-wm-arthur) [inline go/store/prolly/tree/immutable_tree.go:29]: contents

File: go/store/prolly/tree/immutable_tree.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+// tree, returning the root node or an error if applicable. |chunkSize|
+// fixes the split size of leaf and intermediate node chunks.
+func buildImmutableTree(ctx context.Context, r io.Reader, ns NodeStore, S message.Serializer, chunkSize int) (Node, error) {
+	if chunkSize <= 1 {
```
Comment (andrew-wm-arthur) [inline go/store/prolly/tree/immutable_tree.go:33]: 1 byte is invalid?

File: go/store/prolly/tree/immutable_tree_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+				}
+				return nil
+			})
+
+			require.Equal(t, expLevel, root.Level())
+			require.Equal(t, expSum, sum)
+			require.Equal(t, tt.inputSize, byteCnt)
+			require.Equal(t, expUnfilled, unfilledCnt)
+			require.Equal(t, expSubtrees, root.getSubtreeCounts())
+		})
```
Comment (andrew-wm-arthur) [inline go/store/prolly/tree/immutable_tree_test.go:135]: you should add a validation step here that asserts we can round-trip `buf`

**Review by andrew-wm-arthur — APPROVED** [review]
Generally looks good, tree construction logic appears correct. 

The `ProllyTreeNode` flatbuffers message, and the `Serialize` interface are both oriented towards `Tuple` indexes. You should follow up on this PR to create a dedicated message and serialization path. It might also be worthwhile to move blobs into their own package.

File: go/store/prolly/tree/immutable_tree.go (hunk 2 above)
Comment (max-hoffman) [reply go/store/prolly/tree/immutable_tree.go:33]: 1 byte is OK and tested, chunkSize 1 causes infinite recursion (can't be tree)

File: go/store/prolly/tree/immutable_tree_test.go (hunk 3 above)
Comment (max-hoffman) [reply go/store/prolly/tree/immutable_tree_test.go:135]: `sum` validates the contents were written as expected, I was planning on round-tripping in the next PR that adds a struct for read/writing blobs at the tuple layer

**Pull request merged**